submissions/BlueLamantine/friends app/js/main.js

```diff
@@ -0,0 +1,329 @@
+        <label for="nameAZ" class="radio-lable">A - Z</label>
+        <input type="radio" id="nameZA" name="sort-name">
+        <label for="nameZA" class="radio-lable">Z - A</label>
+        <input type="radio" id="nameAll" name="sort-name">
+        <label for="nameAll" class="radio-lable">default</label>
+        </div>
+      </div>
+      
+      <div class="filter filter-age">
+        <p>sort by age</p>
+        <div class="radio-wrapper">
+        <input type="radio" id="age09" name="sort-age">
+        <label for="age09" class="radio-lable">young - old</label>
+        <input type="radio" id="age90" name="sort-age">
+        <label for="age90" class="radio-lable">old - young</label>
+        <input type="radio" id="ageAll" name="sort-age">
+        <label for="ageAll" class="radio-lable">default</label>
+        </div>
+      </div>
+
+      
+      <div class="filter filter-search">
+        <p>Search user by email</p>
+        <input id="nameInput" type="text" size="30" class="searchEmail icon search">
+      </div>
+    `;
+  }
+
+  getFilters() {
+    this.filtersContainer.insertAdjacentHTML('beforeend', this.getHtmlFilter());
+  }
+}
+
+class Filter {
+  constructor() {
+    this.profilesContainer = document.querySelector('.profile-cards');
+    new Render().getFilters();
+    this.filtersPanel = document.querySelector('#filters');
+    this.searchInput = document.querySelector('#nameInput');
+    this.startFilter();
+    this.sortByNameTypes = {
+      type1: 'nameAZ',
+      type2: 'nameZA',
+      type3: 'nameAll',
+    };
+    this.sortByAgeTypes = {
+      type1: 'age09',
+      type2: 'age90',
+      type3: 'ageAll',
+    };
+  }
+
+  unckeckedRadioButton(nodeName) {
+    const activeRadioButtons = document.querySelectorAll(
+      'input[type=radio]:checked'
+    );
+    const checkedRadio = Array.from(activeRadioButtons).find(
+      node => node.name === nodeName
+    );
+    if (checkedRadio) checkedRadio.checked = false;
+  }
+
+  selectSort(selected) {
+    const profileNames = document.querySelectorAll('.user-mail');
+    this.searchInput.addEventListener('input', ({ target }) => {
+      let filter = target.value.toUpperCase();
+      profileNames.forEach(nameNode => {
+        if (nameNode.textContent.toLocaleUpperCase().startsWith(filter)) {
+          nameNode.closest('.user-card').hidden = false;
+        } else {
+          nameNode.closest('.user-card').hidden = true;
+        }
+      });
+    });
+
+    const sort = {
+      'sort-gender': () => this.sortByGender(selected.id),
+      'sort-name': () => {
+        this.unckeckedRadioButton('sort-age');
+        this.sortByContent('.first-name', selected.id, this.sortByNameTypes);
+      },
+      'sort-age': () => {
+        this.unckeckedRadioButton('sort-name');
+        this.sortByContent('.age', selected.id, this.sortByAgeTypes);
+      },
+      default: () => false,
+    };
+    return (sort[selected.name] || sort['default'])();
```

Comment:
`sort['default']` no need for array notation here

Reply:
Point of this function is to provide two kind of sort. One includes different sorts (by name, by age...) and second with search by email. So when we choose ```search``` our ```const sort``` used  variety ```default```: () => false. ```sort['default']``` is necessary,  that will be error without it!

Reply:
Sorry, I have confused you. I meant bracket notation - https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Property_Accessors

Also, if I understand you correctly, default sort is by email?

If so, why function called default then?

Reply:
Just a moment's inattention.  Now it is ```return (sort[selected.name] || sort.sortByEmail)();```